### PR TITLE
refactor sw360 workflow items

### PR DESCRIPTION
**Pull request by neubs-bsi**
extract methods from process to clean code a bit.
this makes reading the code easier.

@bs-ondem 

*Type of change*:  improvements

All tests pass

### CODE REVIEW

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java [hunk 1]
```diff
@@ -45,35 +40,41 @@ public SW360UpdaterImpl(SW360MetaDataUpdater sw360MetaDataUpdater,
...
-                        && sw360ReleaseFinal.get_Links().getSelf() != null
-                        && !sw360ReleaseFinal.get_Links().getSelf().getHref().isEmpty()) {
-                    Map<Path, SW360AttachmentType> attachments = SW360AttachmentAdapterUtils.getAttachmentsFromArtifact(artifact);
-                    if (!attachments.isEmpty()) {
-                        sw360ReleaseFinal = sw360MetaDataUpdater.uploadAttachments(sw360ReleaseFinal, attachments);
-                    }
-                }
-                releases.add(sw360ReleaseFinal);
-            } catch(ExecutionException e) {
-                LOGGER.error("Release will not be created in SW360. Reason: {}", e.getMessage());
```
Comment (bs-ondem) [inline modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java]: Where are these logs in the refactoring?

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java [hunk 2]
```diff
@@ -45,35 +40,41 @@ public SW360UpdaterImpl(SW360MetaDataUpdater sw360MetaDataUpdater,
...
-                LOGGER.debug("Error: ", e);
-            }
+            SW360Release sw360ReleaseFinal = artifactToReleaseInSW360(artifact);
```
Comment (bs-ondem) [inline modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java]:
```suggestion
        List<SW360Release> releases = intermediates.stream()
                .map(this::artifactToReleaseInSW360)
                .collect(Collectors.toList());
```

Comment (neubs-bsi) [reply modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java]: @bs-ondem does this seem suitable to you?
```
List<SW360Release> releases = intermediates.stream()
                .map(artifact -> {
                    try {
                        return artifactToReleaseInSW360(artifact);
                    } catch (ExecutionException e) {
                        LOGGER.error("Release will not be created in SW360. Reason: {}", e.getMessage());
                        return null;
                    }
                })
                .filter(Objects::nonNull)
                .collect(Collectors.toList());
```

Comment (bs-ondem) [reply modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java]: Hmmm... my suggestion was for your changes without the logging.. So in this case you can keep the for loop and put the logging inside the loop... I don't like streams with try-blocks

**Pull request merged**